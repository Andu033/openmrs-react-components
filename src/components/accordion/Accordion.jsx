import React from 'react';
import PropTypes from 'prop-types';
import '../../../assets/css/accordion.css';
import Arrow from '../../../assets/images/arr-right.svg';

class Accordion extends React.Component {
  state = {
    isVisible: this.props.open === undefined ? false : this.props.open,
  };

  render() {
    return (
      <div className={`accordion ${this.props.border ? 'border' : ''}`}>
        <div
          className="header"
          onClick={() => {
            this.setState(() => ({ isVisible: !this.state.isVisible }));
          }}
          role="button"
          tabIndex={0}
        >
          <a>
            <span>
              <img
                className={`${this.state.isVisible ? 'rotate90' : ''}`}
                height="12px"
                src={Arrow}
                width="12px"
              />
            </span>&nbsp;&nbsp;
            {this.props.title}
          </a>
        </div>
        <div className={`content ${!this.state.isVisible ? 'close' : 'open'}`}>
          {this.props.children}
        </div>
      </div>
    );
  }
}


Accordion.propTypes = {
  border: PropTypes.bool,
  children: PropTypes.node.isRequired,
  open: PropTypes.bool,
  title: PropTypes.string.isRequired,
};

Accordion.defaultProps = {
  open: false,
  border: false,
};

export default Accordion;
