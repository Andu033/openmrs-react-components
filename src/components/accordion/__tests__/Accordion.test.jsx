import React from 'react';
import renderer from 'react-test-renderer';
import Accordion from '../Accordion';

let props;
let mountedComponent;

const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = renderer.create(<Accordion {...props} />);
  }
  return mountedComponent;
};

describe('Component: Accordion', () => {
  beforeEach(() => {
    props = {
      open: true,
      title: 'Header',
      children: <h1>I love this</h1>
    };
    mountedComponent = undefined;
  });

  it('renders properly', () => {
    const component = getComponent();
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('calls the onclick', () => {
    const renderedComponent = getComponent().root;
    renderedComponent.findByProps({ className: "header" }).props.onClick();
    expect(renderedComponent.instance.state.isVisible).toEqual(false);
  });
});
