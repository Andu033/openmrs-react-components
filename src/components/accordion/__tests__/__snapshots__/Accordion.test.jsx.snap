// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Accordion renders properly 1`] = `
<div
  className="accordion "
>
  <div
    className="header"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    <a>
      <span>
        <img
          className="rotate90"
          height="12px"
          src="test-file-stub"
          width="12px"
        />
      </span>
        
      Header
    </a>
  </div>
  <div
    className="content open"
  >
    <h1>
      I love this
    </h1>
  </div>
</div>
`;
